sinks/signalfx/signalfx.go

```diff
@@ -208,69 +210,36 @@ func (sfx *SignalFxSink) Flush(ctx context.Context, interMetrics []samplers.Inte
 	return err
 }
 
-// FlushEventsChecks sends events to SignalFx. It does not support checks.
+var successSpanTags = map[string]string{"sink": "signalfx", "results": "success"}
+var failureSpanTags = map[string]string{"sink": "signalfx", "results": "failure"}
+
+// FlushOtherSamples sends events to SignalFx. It does not support checks.
 func (sfx *SignalFxSink) FlushOtherSamples(ctx context.Context, samples []ssf.SSFSample) {
 	span, _ := trace.StartSpanFromContext(ctx, "")
 	defer span.ClientFinish(sfx.traceClient)
-
 	for _, sample := range samples {
-
-		if _, ok := sample.Tags[dogstatsd.EventIdentifierKey]; !ok {
-			// This isn't an event, just continue
-			continue
-		}
-
-		dims := map[string]string{}
-
-		// Copy common dimensions in
-		for k, v := range sfx.commonDimensions {
-			dims[k] = v
-		}
-		// And hostname
-		dims[sfx.hostnameTag] = sfx.hostname
-
-		for k, v := range sample.Tags {
-			if k == dogstatsd.EventIdentifierKey {
-				// Don't copy this tag
-				continue
+		switch ddOtherSampleKind(sample.Tags) {
+		case ddSampleEvent:
+			ev := sfx.reportEvent(&sample)
+			// TODO: Split events out the same way as points
+			sfx.defaultClient.AddEvents(ctx, []*event.Event{&ev})
+			// report evt
+		case ddSampleServiceCheck:
+			err := sfx.convertServiceCheck(&sample)
+			spanTags := successSpanTags
+			if err != nil {
+				spanTags = failureSpanTags
 			}
-			dims[k] = v
-		}
-
-		for k := range sfx.excludedTags {
-			delete(dims, k)
-		}
-		name := sample.Name
-		if len(name) > EventNameMaxLength {
-			name = name[0:EventNameMaxLength]
-		}
-		message := sample.Message
-		if len(message) > EventDescriptionMaxLength {
-			message = message[0:EventDescriptionMaxLength]
-		}
-		// Datadog requires some weird chars to do markdown… SignalFx does not so
-		// we'll chop those out
-		// https://docs.datadoghq.com/graphing/event_stream/#markdown-events
-		message = strings.Replace(message, "%%% \n", "", 1)
-		message = strings.Replace(message, "\n %%%", "", 1)
-		// Sometimes there are leading and trailing spaces
-		message = strings.TrimSpace(message)
-
-		ev := event.Event{
-			EventType:  name,
-			Category:   event.USERDEFINED,
-			Dimensions: dims,
-			Timestamp:  time.Unix(sample.Timestamp, 0),
-			Properties: map[string]interface{}{
-				"description": message,
-			},
+			span.Add(ssf.Count(sinks.ServiceCheckConversionCount, 1.0, spanTags))
```

Comment:
I think this would probably be better to count internally (using a variable) first, then report that value... I don't think it'll blow up internal metrics processing, but seems prudent to do still.

Reply:
Okay. I had considered that but thought that the span might roll it up ahead of time so I just enqueued. Will change.